src/libraries/Common/src/Internal/Cryptography/Helpers.cs

```diff
@@ -30,11 +30,11 @@ internal static partial class Helpers
             return (byte[])(src.Clone());
         }
 
-        public static int GetPaddingSize(this SymmetricAlgorithm algorithm)
+        public static int GetPaddingSize(this SymmetricAlgorithm algorithm, CipherMode mode, int feedbackSizeBits)
         {
             // CFB8 does not require any padding at all
             // otherwise, it is always required to pad for block size
-            if (algorithm.Mode == CipherMode.CFB && algorithm.FeedbackSize == 8)
+            if (mode == CipherMode.CFB && feedbackSizeBits == 8)
                 return 1;
 
             return algorithm.BlockSize / 8;
```

Comment:
Would it make sense to take the block size parameter separately and get rid of the algorithm/this parameter?  Then nothing could sneak its way in.

Reply:
Probably; I wanted to clean this all up a bit when I get to the CFB implementation. There's a bit of weirdness of so many things being passed around... some of it is in bits, some of it is in bytes...